vfs/cache.go

```diff
@@ -270,12 +270,63 @@ func (c *cache) exists(name string) bool {
 	if err != nil {
 		return false
 	}
-	if fi.IsDir() {
+	// checks for non-regular files (e.g. directories, symlinks, devices, etc.)
+	if !fi.Mode().IsRegular() {
 		return false
 	}
 	return true
 }
 
+// renames the file in cache
+func (c *cache) rename(name string, newName string) bool {
+	osOldPath := c.toOSPath(name)
+	osNewPath := c.toOSPath(newName)
+	sfi, err := os.Stat(osOldPath)
+	if err != nil {
+		return false
+	}
+	if !sfi.Mode().IsRegular() {
+		// cannot copy non-regular files (e.g., directories, symlinks, devices, etc.)
+		fs.Errorf("rename: non-regular source file %s (%q)", sfi.Name(), sfi.Mode().String())
```

Comment:
... so instead of fs.Errorf(...); return false

do
```go
return errors.Errorf("rename: non-regular source file %s (%q)", sfi.Name(), sfi.Mode().String()))
```